frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/create-obc.tsx

```diff
@@ -0,0 +1,155 @@
+  const dropDownUnits = {
+    MiB: 'MiB',
+    GiB: 'GiB',
+    TiB: 'TiB',
+  };
+
+  React.useEffect(() => {
+    const obj: K8sResourceKind = {
+      apiVersion: `${NooBaaObjectBucketClaimModel.apiGroup}/${
+        NooBaaObjectBucketClaimModel.apiVersion
+      }`,
+      kind: NooBaaObjectBucketClaimModel.kind,
+      metadata: {
+        namespace,
+      },
+      spec: {
+        ssl: false,
+      },
+    };
+    if (state.scName) {
+      obj.spec.storageClassName = state.scName;
+    }
+    if (state.name) {
+      obj.metadata.name = state.name;
+      obj.spec.generateBucketName = state.name;
+    } else {
+      obj.metadata.generateName = 'true';
+      obj.metadata.generateBucketName = true;
+    }
+    dispatch({ type: 'setPayload', payload: obj });
+  }, [namespace, state.name, state.scName]);
+
+  const save = (e: React.FormEvent<EventTarget>) => {
+    e.preventDefault();
+    dispatch({ type: 'setProgress' });
+    k8sCreate(NooBaaObjectBucketClaimModel, state.payload)
+      .then((resource) => {
+        dispatch({ type: 'unsetProgress' });
+        history.push(resourceObjPath(resource, referenceFor(resource)));
+      })
+      .catch((err) => {
+        dispatch({ type: 'setError', message: err.message });
+        dispatch({ type: 'unsetProgress' });
+      });
+  };
+
+  return (
+    <div className="co-m-pane__body co-m-pane__form">
+      <Helmet>
+        <title>Create Object Bucket Claim</title>
+      </Helmet>
+      <h1 className="co-m-pane__heading co-m-pane__heading--baseline">
+        <div className="co-m-pane__name">Create Object Bucket Claim</div>
+        <div className="co-m-pane__heading-link">
+          <Link to={`/k8s/ns/${namespace}/objectbucketclaims/~new`} id="yaml-link" replace>
+            Edit YAML
+          </Link>
+        </div>
+      </h1>
+      <form className="co-m-pane__body-group" onSubmit={save}>
+        <div>
+          <div className="form-group">
+            <label className="control-label" htmlFor="obc-name">
+              Object Bucket Claim Name
+            </label>
+            <div className="form-group">
+              <input
+                className="pf-c-form-control"
+                type="text"
+                onChange={(e) => dispatch({ type: 'setName', name: e.currentTarget.value })}
+                placeholder="my-object-bucket"
+                aria-describedby="obc-name-help"
+                id="obc-name"
+                name="obcName"
+                pattern="[a-z0-9](?:[-a-z0-9]*[a-z0-9])?"
+              />
+              <p className="help-block" id="obc-name-help">
+                If not provided, a generic name will be generated.
+              </p>
+            </div>
+            <div className="form-group">
+              <StorageClassDropdown
+                onChange={(sc) =>
+                  dispatch({ type: 'setStorage', name: _.get(sc, 'metadata.name') })
+                }
+                id="storageclass-dropdown"
```

Comment:
@bipuladh Are we using `storageclass-dropdown` id anywhere?

Reply:
ack